echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py

```diff
@@ -0,0 +1,1091 @@
+    consistent for all groups. This is necessary because
+    not all groups have the same attributes (some are
+    missing them too) for the ``frequency`` variable.
+    This variable is used to set the variable
+    ``frequency_nominal`` later on.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    for grp_path in ed_obj.group_paths:
+
+        if "frequency" in list(ed_obj[grp_path].coords):
+
+            # creates consistent frequency attributes
+            # NOTE: the attribute standard_name will be removed in `frequency_nominal`
+            ed_obj[grp_path]["frequency"] = ed_obj[grp_path].frequency.assign_attrs(
+                {
+                    "long_name": "Transducer frequency",
+                    "standard_name": "sound_frequency",
+                    "units": "Hz",
+                    "valid_min": 0.0,
+                }
+            )
+
+
+def _reorganize_beam_groups(ed_obj):
+    """
+    Maps Beam --> Sonar/Beam_group1 and Beam_power --> Sonar/Beam_group2.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    # Map Beam --> Sonar/Beam_group1
+    if "Beam" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam"])
+        ed_obj._tree["Sonar/Beam"].name = "Beam_group1"
+
+    # Map Beam_power --> Sonar/Beam_group2
+    if "Beam_power" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam_power"])
+        ed_obj._tree["Sonar/Beam_power"].name = "Beam_group2"
+
+
+def get_channel_id(ed_obj, sensor):
+    """
+    Returns the channel_id for all non-unique frequencies.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        The sensor used to create the v0.5.x file.
+
+    Returns
+    -------
+    A datarray specifying the channel_ids with dimension frequency.
+    """
+    if sensor == "AZFP":
+
+        # create frequency_nominal variable
+        freq_nom = ed_obj["Sonar/Beam_group1"].frequency
+
+        # create unique channel_id for AZFP
+        freq_as_str = (freq_nom / 1000.0).astype(int).astype(str).values
+        channel_id_str = [
+            str(ed_obj["Sonar"].sonar_serial_number) + "-" + freq_as_str[i] + "-" + str(i + 1)
+            for i in range(len(freq_as_str))
+        ]
+        channel_id = xr.DataArray(
+            data=channel_id_str, dims=["frequency"], coords={"frequency": freq_nom}
+        )
+
+    else:
+
+        channel_id = xr.concat(
+            [child.ds.channel_id for child in ed_obj._tree["Sonar"].children], dim="frequency"
```

Comment:
I haven't tried to evaluate this myself, but since dim is set to `frequency` and `xr.concat` is used directly, I am curious if this works in the case with duplicated frequencies. BUT, we won't have a v0.5.x file with such characteristics because it won't convert, so this is moot.

Reply:
> BUT, we won't have a v0.5.x file with such characteristics because it won't convert, **so this is moot.**

Indeed

Reply:
I am unsure what will happen with repeated frequencies. As to not spend time on this, I will not look into this right now (since we will not encounter this scenario).